### ISSUE TITLE
coherent asset style

### ISSUE THREAD

**Issue by Katschej** [Uriopass/Egregoria#93]
Right now there are quite a few different asset styles in the game. While this is good to compare and experiment with different styles, I think the game would look a lot better if it had a coherent style.

I would categorise the assets currently in the game like this:

## "flat"

![flat](https://github.com/Uriopass/Egregoria/assets/141503035/a90e080b-2938-440a-9284-8a828eb36d76)

The vast majority of the game looks like this (cars, trains, predestrians, etc). My example is very flat. There could for example be a normal, metallic or roughness texture (or different materials) for aditional detail, while still keeping the flat look (as is done with the trains).

## "pixelated"

![pixel](https://github.com/Uriopass/Egregoria/assets/141503035/3de1595a-7c23-4756-b817-6025e0021caf)

Only the bakery looks like this. It is very quick to modell and quite a novel look. 

## "textured"

![textured](https://github.com/Uriopass/Egregoria/assets/141503035/79892495-0d18-46c2-80d2-8a52af9b8187)

The solar panels, flour factory, coal power plant, and wheat farm look like this. Since most of the game is spent relativly far away from the modells, there actually isn't that much detail needed, to look relativly convincing / good.

## 
The oil pump in the "flat" version was made by replacing the texture from "textured" with the palette.png and adjusting the UVs. Going from "textured" to "flat" is only really a matter of texture. That means the currently "flat" assets should be relatively easily converted (maybe a bit more detail is needed, for example side mirrors for the cars, etc.) and vice versa.

"textured" version (my personal favorite) in game:
![screen](https://github.com/Uriopass/Egregoria/assets/141503035/826e440b-43eb-4be7-8423-6da6c3fdd045)

blend file: [side_by_side.zip](https://github.com/Uriopass/Egregoria/files/12269568/side_by_side.zip)
(I needed to duplicate some faces and flip the normal so that they would be visible form both sides in the game)

**Comment by Zireael07** [comment]
My vote is for textured (textures of this sort are relatively easy to find and edit* yourself, it's what I do for my own racer game)

* because nearly all of the textures I use are Creative Commons licensed which allows editing/remixing

**Comment by Uriopass** [comment]
Textured is absolutely the goal! I went for flat/pixelated assets for some of them because I didn't have the time to make proper models. The bakery is very bad, it is a very old model.
The trees can be textured but they must stay low poly until we get Tree Impostors (#89). In your examples I think they are.

Thank you for your work and the models, it looks very good! The trees could probably be a bit greener (lighter), the best would be to have different colors depending on the region and variations but that's a whole new thing

**Comment by Uriopass** [comment]
> (I needed to duplicate some faces and flip the normal so that they would be visible form both sides in the game)

I used to support double sided materials but I was very worried about this since by default blender doesn't enable backface culling so it would incur big performance loss for nothing.
I could add a custom property to have it for the few models that need it.

**Comment by Katschej** [comment]
> Textured is absolutely the goal! I went for flat/pixelated assets for some of them because I didn't have the time to make proper models. The bakery is very bad, it is a very old model. The trees can be textured but they must stay low poly until we get Tree Impostors (#89). In your examples I think they are.

That's good to know. I think I will do some more models (the missing ones are probably the priority I'd assume). The tree was just the pine with a different texture. I should be able to make it a bit greener / lighter. Is there any performance disadvantage to alpha channels?

> Thank you for your work and the models, it looks very good! The trees could probably be a bit greener (lighter), the best would be to have different colors depending on the region and variations but that's a whole new thing

Thank you! I could try to make some other tree types with ~ 50-100 tries (or is that to high?).  Although I probably can't implement them, apart from replacing the pine.

> I used to support double sided materials but I was very worried about this since by default blender doesn't enable backface culling so it would incur big performance loss for nothing. I could add a custom property to have it for the few models that need it.

I don't think you need that, duplicating faces isn't that time consuming. Although I don't know if there is any performance disadvantage to that (over just duplicating the faces). If there is it might be usefull.

**Comment by Uriopass** [comment]
> Is there any performance disadvantage to alpha channels?

Not really, but the engine doesn't really support transparency in the models, only alpha cutouts but that's fine for foliage. (so like transparent/opaque is ok but nothing in between)

> ~ 50-100 tries (or is that to high?)

Current model is 36 tris but I'd say 50-100 is fine. Better trees can also mean that less density still looks better.

> Although I don't know if there is any performance disadvantage to that

There is a small performance disadvantage since the gpu needs to process a bit more vertices rather than just disabling backface culling, but it's more than fine.

**Comment by Uriopass** [comment]
I'm not sure how to ratify the "textured" decision. Maybe this issue is enough?

**Comment by Katschej** [comment]
I think it is.
